client/e2e-test/script-test.js

```diff
@@ -78,25 +79,36 @@ const datasetName = Date.now().toString();
       return Promise.resolve(found.getAttribute('data-test-id'));
     });
     console.log(`ID extracted: ${id}\n`);
-
+    let pending;
+    do {
+      pending = await page.$(`[data-test-name="${datasetName}"] [data-test-id="pending"]`);
+    } while (pending);
```

Comment:
infinite loop?

Reply:
The data-test-id "pending" disappears when the dataset is completely created and available. Then, the variable pending will be undefined, and the JavaScript API says that this will return false: 
https://developer.mozilla.org/es/docs/Glossary/Truthy
So the loop will end in that moment.

Reply:
So an infinite loop if the "pending" does not disappear because something is broken.

Reply:
Now we added a timeout for the loop.